src/Controller/DemoController.php

```diff
@@ -36,55 +36,68 @@ final class DemoController extends AbstractController
+        $this->fileDiffCleaner = $fileDiffCleaner;
     }
 
     /**
-     * @Route(path="demo/{id}", name="demo_detail", methods={"GET", "POST"})
+     * @Route(path="demo/{rectorRun}", name="demo_detail", methods={"GET", "POST"})
      * @Route(path="demo", name="demo", methods={"GET", "POST"})
      */
-    public function __invoke(Request $request): Response
+    public function __invoke(Request $request, ?RectorRun $rectorRun = null): Response
     {
-        $formData = new RectorRunFormData();
-        $rectorRun = null;
-        $rectorRunId = $request->attributes->get('id');
-
-        $demoFileContent = FileSystem::read(__DIR__ . '/../../data/DemoFile.php');
-        $demoConfig = FileSystem::read(__DIR__ . '/../../data/demo-config.yaml');
-
-        $formData->setContent($demoFileContent);
-        $formData->setConfig($demoConfig);
-
-        if ($rectorRunId) {
-            $rectorRun = $this->rectorRunRepository->get(Uuid::fromString($rectorRunId));
-
-            $formData->setContent($rectorRun->getContent());
-            $formData->setConfig($rectorRun->getConfig());
-        }
+        $formData = $this->createRectorRunFormData($rectorRun);
 
-        $form = $this->createForm(RectorRunFormType::class, $formData);
+        $form = $this->createForm(DemoFormType::class, $formData);
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             return $this->processFormAndReturnRoute($form);
         }
 
         return $this->render('homepage/demo.twig', [
-            'form' => $form->createView(),
-            'rector_run' => $rectorRun,
+            'demo_form' => $form->createView(),
+            'rector_run' => $rectorRun ?? null,
         ]);
     }
 
+    private function createRectorRunFormData(?RectorRun $rectorRun): DemoFormData
+    {
+        $formData = new DemoFormData();
+
+        if ($rectorRun) {
+            $formData->setContent($rectorRun->getContent());
+            $formData->setConfig($rectorRun->getConfig());
+        } else {
+            // default values
+            $demoFileContent = FileSystem::read(__DIR__ . '/../../data/DemoFile.php');
+            $demoConfig = FileSystem::read(__DIR__ . '/../../data/demo-config.yaml');
+
+            $formData->setContent($demoFileContent);
+            $formData->setConfig($demoConfig);
+        }
+
+        return $formData;
+    }
+
     private function processFormAndReturnRoute(FormInterface $form): RedirectResponse
     {
-        /** @var RectorRunFormData $formData */
-        $formData = $form->getData();
-        $config = $formData->getConfig();
+        /** @var DemoFormData $rectorRunFormData */
+        $rectorRunFormData = $form->getData();
+        $config = $rectorRunFormData->getConfig();
 
-        $rectorRun = new RectorRun(Uuid::uuid4(), new DateTimeImmutable(), $config, $formData->getContent());
+        $rectorRun = $this->createRectorRun($config, $rectorRunFormData);
```

Comment:
Basically just renamed `RectorRun*` to `Demo*`, for consistency with Form, Controller and template